client/capstone-app/utils.js

```diff
@@ -90,5 +90,69 @@ function midpoint(coord1, coord2) {
   return [midpointLat, midpointLon];
 }
 
+  // Helper function to flatten nested arrays
+function parseRouteData(data) {
```

Comment:
Can we clarify this comment a little more? What format is the route data initially in and why are we flattening it?

Reply:
This has been addressed in future PRs! 😁